test/mainnet/configurer.spec.sol

```diff
@@ -233,75 +149,40 @@ contract VaultConfigureUpgradeTest is TestHelper, MainnetActors {
         );
     }
 
-    function _emptyVault() internal {
-        address[] memory assets = vault.getAssets();
-        assertEq(assets.length, 11, "Should have 11 assets");
-
-        // empty vault so we can test each asset in isolation
-        for (uint256 i = 0; i < assets.length; i++) {
-            uint256 balanceBefore = IERC20(assets[i]).balanceOf(address(vault));
-            if (balanceBefore > 0) {
-                vm.startPrank(address(vault));
-                IERC20(assets[i]).transfer(address(this), balanceBefore);
-                vm.stopPrank();
-            }
-        }
-        vault.processAccounting();
-
-        assertEq(vault.totalAssets(), 0);
-    }
+    function test_donate_assets(uint256 donationAmount) public {
+        vm.assume(donationAmount > 1e8);
+        vm.assume(donationAmount < 1_000 ether);
 
-    function test_donate_assets(uint256 donationAmount, uint8 i) public {
         address[] memory assets = vault.getAssets();
         assertEq(assets.length, 11, "Should have 11 assets");
-        vm.assume(i < assets.length);
 
-        vm.assume(donationAmount > 1e8);
-        if (assets[i] == MC.STETH) {
-            vm.assume(donationAmount < 1000 ether);
-        } else {
-            vm.assume(donationAmount < 1e5 ether);
+        for (uint256 i = 0; i < assets.length; i++) {
+            _test_donate_single_asset(assets[i], donationAmount);
         }
-
-        // empty vault so we can test each asset in isolation
-        _emptyVault();
```

Comment:
this test had some errors such as emptying the vault mid-test which creates a very particular testing environment (eg . totalShares are 1 million, totalAssets is 0) which doesn't reflect well what's being tested